packages/grid/_modules_/grid/components/containers/GridRootStyles.ts

```diff
@@ -121,8 +117,18 @@ export const useStyles = makeStyles(
         '& .MuiDataGrid-colCellCenter .MuiDataGrid-colCellTitleContainer': {
           justifyContent: 'center',
         },
-        '& .MuiDataGrid-colCellRight .MuiDataGrid-colCellTitleContainer': {
-          justifyContent: 'flex-end',
+        '& .MuiDataGrid-colCellCenter.MuiDataGrid-withColumnMenu .MuiDataGrid-colCellTitleContainer': {
+          transform: 'translateX(-10px)',
```

Comment:
Another option is to add `position: absolute` to the menu so it's not considered as part of the div when aligning to the center. This alternative also requires to apply a background or the header name will go over the column menu icon.

![image](https://user-images.githubusercontent.com/42154031/114743605-385d8000-9d23-11eb-8e71-44d8a663c70c.png)